dmitrizagidulin/solid-server-unstable
# OIDC Integration - Second Iteration

- WIP (fix standard warnings)
- WIP (moving oidcConfig to config.json)
- Add OIDC Issuer discovery link header on OPTIONS
- Refactor OIDC client/handler code
- Bump dep version, fix gitignore
- Integrate Sign In w WebID page with OIDC login
- Fix OIDC create user functionality
- Fix storing redirectTo URL in session
- Remove unneeded options obj (oidc.js)
- WIP (create account using webid as OIDC _id)
- Fix extraneous oidcIssuerHeader
- WIP (fix response types)
- Fix token() params
- Fix authCallback()
- WIP (switch to implicit workflow)
- WIP (fix registration configs)
- wip - Switch to authz code flow
- Move createOIDCUser to oidc-rp-client (from identity-provider)
- Implement OIDC /signout support
- Do not create a Solid account folder if OIDC user creation fails
- WIP - signin session / resume workflow
- Signin api: send 500 error if oidc client not initialized
- Remove invalid test
- Implement user creation via api
- Add username & pw auth logic to signin.js
- Add bcrypt compilation Travis CI directives
- Mount OIDC Identity Provider API into solid server routes
- Initialize multi-rp client store collections
- Change oidc packages to use github links
- Disable the certificate test for the time being
- Derive session cookie domain from root uri
- Bump oidc-op-express dep
- Output auth method at startup
- Implement injected Provider logout() behavior
- Remove legacy signout logic
- Refactoring OIDC initialization from server config in create-app
- Simplify createApp()
- Move unit tests to their own test/unit/ folder
- Refactor lib/api/authn/ files
- Move tests to test/unit/ and test/integration/
- Implement CreateOidcAccountRequest and unit tests
- Add OIDC account creation integration tests
- Extract OidcManager code to external lib
- Fix Provider Discover integration tests
- Fix existing Login workflow integration test
- Implement DiscoverProviderRequest, refactor
- Extract login logic to LoginByPasswordRequest
- Unit tests for LoginByPasswordRequest
- Implement login via either username or webid
- Add /login integration tests
- Normalize webid if needed during provider discovery
- Render Obtain Consent handlebars view
- Save provider config on startup, install nyc / code coverage
- Move options handler to before authentication or account api
- Ensure local client registered on startup
